rust/worker/src/server.rs

```diff
@@ -116,7 +117,7 @@ impl WorkerServer {
         Ok(())
     }
 
-    pub(crate) fn set_dispatcher(&mut self, dispatcher: Box<dyn Receiver<TaskMessage>>) {
+    pub(crate) fn set_dispatcher(&mut self, dispatcher: ComponentHandle<Dispatcher>) {
```

Comment:
I'm fine keeping this as `Box<dyn Receiver<TaskMessage>>` if that style is preferred

Reply:
(actually I guess the next PR would result in this needing to be `ComponentHandle<Dispatcher>` so we know the result type, unless we want to attach the result type to the message rather than the handler.)

Reply:
i prefer this! nice cleanup